src/internal/storage/fileset/index/reader.go

```diff
@@ -56,11 +57,15 @@ func (r *Reader) Iterate(ctx context.Context, cb func(*Index) error) error {
 	if r.topIdx == nil {
 		return nil
 	}
+	peek := r.peek
 	traverseCb := func(idx *Index) (bool, error) {
-		if atEnd(idx.Path, r.filter) {
-			return false, errutil.ErrBreak
-		}
 		if idx.File != nil {
+			if atEnd(idx.Path, r.filter) {
```

Comment:
Yeah, we want to check the first file index after the end of the path range. This may involve traversing additional range indexes if the end of the path range lines up with a chunk boundary.